### PR TITLE
Replace int as underlying type in IntLitExpr and RatLitExpr

### LINKED ISSUE #42: IntLitExpr should have a larger value range
Programs cannot be verified if they contain integers (such as 4294967295) which are larger than the limit of the signed java int type. The internal value held by IntLitExpr should be replaced with long (at least for the time being). This may also affect certain DSL parsers.

### PR DESCRIPTION
**Pull request by as3810t**
This pull request replaces int as underlying type in IntLitExpr and RatLitExpr, subsequently closes #42 .

### CODE REVIEW

**Comment by as3810t** [comment]
As of now, theta builds successfully, however more integration tests should be run to evaluate the changes.

File: subprojects/core/src/test/java/hu/bme/mit/theta/core/utils/ExprSimplifierTest.java [hunk 1]
```diff
@@ -269,6 +270,7 @@ public void testIntAdd() {
 		assertEquals(Add(a, Int(4)), simplify(Add(Int(1), Add(a, Int(3)))));
 		assertEquals(a, simplify(Add(Int(-3), a, Int(3))));
 		assertEquals(Add(a, b, a, b, c), simplify(Add(a, Add(b, Add(a, Add(b, c))))));
+		assertEquals(Int(new BigInteger("4294967294")), simplify(Add(Int(Integer.MAX_VALUE), Int(Integer.MAX_VALUE))));
```
Comment (hajduakos) [inline subprojects/core/src/test/java/hu/bme/mit/theta/core/utils/ExprSimplifierTest.java:273]: Nice

**Review by hajduakos — APPROVED** [review]
Looks good

**Comment by hajduakos** [comment]
I'm just about to run tests. I'll get back when there are results

File: subprojects/core/src/test/java/hu/bme/mit/theta/core/utils/ExprSimplifierTest.java (hunk 1 above)
Comment (as3810t) [reply subprojects/core/src/test/java/hu/bme/mit/theta/core/utils/ExprSimplifierTest.java:273]: Right now `Int` takes either `int` or a `BigInteger`. What do you think, should I implement az overload that takes a `String`, making it easier to create "big" literals?

Comment (hajduakos) [reply subprojects/core/src/test/java/hu/bme/mit/theta/core/utils/ExprSimplifierTest.java:273]: Yes, I think that would be useful. Basically that would just pass `new BigInteger(string)` to the existing constructor, right?

Comment (hajduakos) [reply subprojects/core/src/test/java/hu/bme/mit/theta/core/utils/ExprSimplifierTest.java:273]: By constructor I mean the static factory method.

Comment (as3810t) [reply subprojects/core/src/test/java/hu/bme/mit/theta/core/utils/ExprSimplifierTest.java]: Implemented in b05cd37

**Pull request merged**